velox/functions/sparksql/tests/ArrayPrependTest.cpp

```diff
@@ -0,0 +1,178 @@
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayPrependTest : public SparkFunctionBaseTest {
+ protected:
+  void testExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    auto result = evaluate(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  template <typename T>
+  void testFloats() {
+    static const T kNaN = std::numeric_limits<T>::quiet_NaN();
+    static const T kSNaN = std::numeric_limits<T>::signaling_NaN();
+    {
+      const auto arrayVector = makeNullableArrayVector<T>(
+          {{std::nullopt, 2, 3, std::nullopt, 4},
+           {3, 4, 5, kNaN, 3, 4, kNaN},
+           {kSNaN, 8, 9}});
+      const auto elementVector = makeFlatVector<T>({1, kNaN, kNaN});
+      const auto expected = makeNullableArrayVector<T>({
+          {1, std::nullopt, 2, 3, std::nullopt, 4},
+          {kNaN, 3, 4, 5, kNaN, 3, 4, kNaN},
+          {kNaN, kSNaN, 8, 9},
+      });
+      testExpression(
+          "array_prepend(c0, c1)", {arrayVector, elementVector}, expected);
+    }
+    // Test array_prepend with complex-type elements.
+    {
+      RowTypePtr rowType;
+      if constexpr (std::is_same_v<T, float>) {
+        rowType = ROW({REAL(), VARCHAR()});
+      } else {
+        static_assert(std::is_same_v<T, double>);
+        rowType = ROW({DOUBLE(), VARCHAR()});
+      }
+      using ArrayOfRow = std::vector<std::optional<std::tuple<T, std::string>>>;
+      std::vector<ArrayOfRow> data = {
+          {{{1, "red"}}, {{2, "black"}}, {{3, "green"}}},
+          {{{1, "red"}}, {{2, "black"}}, {{3, "green"}}}};
+      auto arrayVector = makeArrayOfRowVector(data, rowType);
+
+      const auto elementVector =
+          makeConstantRow(rowType, variant::row({kNaN, "blue"}), 2);
+
+      std::vector<ArrayOfRow> expectedData = {
+          {{{kNaN, "blue"}}, {{1, "red"}}, {{2, "black"}}, {{3, "green"}}},
+          {{{kNaN, "blue"}}, {{1, "red"}}, {{2, "black"}}, {{3, "green"}}}};
+      const auto expected = makeArrayOfRowVector(expectedData, rowType);
+      testExpression(
+          "array_prepend(c0, c1)", {arrayVector, elementVector}, expected);
+    }
+  }
+};
+
+// Prepend simple-type elements to array.
+TEST_F(ArrayPrependTest, intArrays) {
+  const auto arrayVector = makeArrayVector<int64_t>(
+      {{1, 2, 3, 4}, {3, 4, 5}, {7, 8, 9}, {10, 20, 30}});
+  const auto elementVector = makeFlatVector<int64_t>({11, 22, 33, 44});
+
+  VectorPtr expected = makeArrayVector<int64_t>({
+      {11, 1, 2, 3, 4},
+      {22, 3, 4, 5},
+      {33, 7, 8, 9},
+      {44, 10, 20, 30},
+  });
+  testExpression(
+      "array_prepend(c0, c1)", {arrayVector, elementVector}, expected);
+}
+
+//// Prepend simple-type elements to array.
```

Comment:
Use // for comments.  Please fix it through.

Reply:
Thanks, resolved :)

Reply:
Please fix all such cases in this file, thanks.